Classes/Command/SupervisorCommandController.php

```diff
@@ -56,7 +49,74 @@ public function createCommand(): void
                 );
             }
         }
+    }
+
+    /**
+     * Starts all programs of configured groups
+     */
+    public function startGroupsCommand(): void
+    {
+        $groups = $this->getGroups();
+        array_walk($groups, [$this, 'runSupervisorCommand'], 'start');
+    }
+
+    /**
+     * Stops all programs of configured groups
+     */
+    public function stopGroupsCommand(): void
+    {
+        $groups = $this->getGroups();
+        array_walk($groups, [$this, 'runSupervisorCommand'], 'stop');
+    }
+
+    /**
+     * Reloads config and starts/stops programs accordingly
+     */
+    public function updateGroupsCommand(): void
+    {
+        $groups = $this->getGroups();
+        array_walk($groups, [$this, 'runSupervisorCommand'], 'update');
+    }
+
+    protected function runSupervisorCommand(Model\Group $group, int $key, string $action): bool
```

Comment:
```
$ supervisorctl help start
start <name>		Start a process
start <gname>:*		Start all processes in a group
start <name> <name>	Start multiple processes or groups
start all		Start all processes
```

If I'm not mistaken here, most supervisorctl actions accept any number of groups and will perform the task on all of them.

So we could use a method signature like this:
```
protected function runSupervisorCommand(string $action, Model\Group ... $groups): bool
```
and get rid of the "array_walk" from above.

This should run through all groups at once instead of waiting for one group to complete.

Reply:
That's very nifty